### ISSUE TITLE
Problem with Swift 3

### ISSUE THREAD

**Issue by frcocoatst** [vishalvshekkar/ArchivingSwiftStructures#1]
I updated to Swift 3 and after conversion it looks like I cannot read the archived struct.
Any idea what I did wrong?

**Comment by frcocoatst** [comment]
[mytry.txt](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/583459/mytry.txt)

**Comment by frcocoatst** [comment]
[Archiv.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/583461/Archiv.zip)

**Comment by vishalvshekkar** [comment]
I haven't yet converted these files in Swift 3.0

Give me some time, I'll look into it.

**Comment by frcocoatst** [comment]
Thanks! Somehow the unarchiving seems to return nil ....

**Comment by vishalvshekkar** [comment]
Yes, I believe there's some change in the folder hierarchy or the way file URLs are handled in Swift 3.0 on a playground.

Can you try archiving it into UserDefaults and unarchiving from it instead of a file?
That should work. Let me know how it goes.

I'll work on conversion and release an update soon.

**Comment by frcocoatst** [comment]
Very strange.
I tried to make a macOS Playground, but get the same problem. The file is definitely at /tmp/moviedata
But reading back fails....
[ArchivingSwiftStructures1.playground.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/585436/ArchivingSwiftStructures1.playground.zip)

**Comment by frcocoatst** [comment]
The same with classes seems to work
[MyNSCoderMacOsPlayground.playground.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/585450/MyNSCoderMacOsPlayground.playground.zip)

**Comment by OptTrader** [comment]
Yeah, I have trouble with conversion also specifically with the the stringappendingstring. Can you also show how to do this with an array of movie objects?

**Comment by frcocoatst** [comment]
Hmmm, no idea what is wrong ...

**Comment by vishalvshekkar** [comment]
Okay, I got the ArchivingSwiftStructures3.playground working in Swift 3. The file's been attached below.

I suspect the problem came up because NSKeyedArchiver and UserDefaults class now use 'Any' type instead of 'AnyObject'. This is causing havoc with class type objects. I'll update the repo with Swift 3 files soon. See if what I've attached suffices for now.

[ArchivingSwiftStructures3_Converted_Swift3.playground.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/610852/ArchivingSwiftStructures3_Converted_Swift3.playground.zip)

**Comment by frcocoatst** [comment]
Works perfect! Thanks a lot!

**Comment by frcocoatst** [comment]
Hi!
Just found one problem (Maybe I'm doing anything wrong). I tried to store an array of structs of NSPoints.
But it fails with "Argument type '[Points]' does not conform to expected type 'Dictionariable'"
[MyArchivePointsPlayground.playground.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/616455/MyArchivePointsPlayground.playground.zip)

Do you have any idea, how to solve this?
Thanks

**Comment by vishalvshekkar** [comment]
If you're using the 3rd type of archiving and unarchiving structs or array of structs, you need to first conform that struct to Dictionariable protocol
Write an extension on NSPoibt or CGPoibt and make it conform to Dictionariable by implementing the two required methods.
Let me know if that helps.

On Mon, Nov 28, 2016 at 7:33 PM, Fritz <notifications@github.com> wrote:
Hi!
Just found one problem (Maybe I'm doing anything wrong). I tried to store an array of structs of NSPoints.
But it fails with "Argument type '[Points]' does not conform to expected type 'Dictionariable'"
MyArchivePointsPlayground.playground.zip [https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/616455/MyArchivePointsPlayground.playground.zip]

Do you have any idea, how to solve this?
Thanks

**Comment by frcocoatst** [comment]
Thanks a lot this helped. 
So I now have two extensions - one for Points structure and one for NSPoint. Right?

[MyArchivePointsPlayground.playground.zip](https://github.com/vishalvshekkar/ArchivingSwiftStructures/files/618688/MyArchivePointsPlayground.playground.zip)

**Comment by vishalvshekkar** [comment]
Awesome.
Yes, that's right. You extend both Points and NSPoint to conform to Dictionariable protocol.
